### ISSUE TITLE
[P2] Rayleigh branch bypasses refractive-index clipping and produces NaNs

### ISSUE THREAD

**Issue by aliakherati** [reflective-org/neuralmie-jax#2]
Review of neuralmie-jax commit `0e709618db38fd350577c1ee4b72c7e4b7f875c7`. The same code is vendored in climate-analytics-lab/jax-gcm#792 (`65e8e78201dae785c35a67a533f829b9d0338b2a`). Reproduced on CPU with JAX 0.11.1.

In `src/neuralmie_jax/neuralmie.py:325-347`, `_bulk_optics()` clips `m_r` and `m_i` for the neural features but passes the previously constructed `ray_args` to the analytic branch. The sphere caller constructs that tuple from the original inputs; the core-shell caller also mixes the original shell indices before `_bulk_optics()` clips them.

Consequently `clip=True` has different semantics in the two branches, and the analytic branch can produce NaNs despite the promised clipping.

```python
import jax
from neuralmie_jax import neuralmie as nm

args = (1e-4, 1e-8, 1.8)
print(nm.sphere_bulk_optics(*args, 1.0, 0.0))
# ke_rho=0, ssa=nan, rayleigh=True
print(nm.sphere_bulk_optics(*args, 1.1, 1e-8))
# ke_rho=0.0012134989, ssa=0.00506285, rayleigh=True
print(nm.coreshell_bulk_optics(*args, 1.0, 0.0, 1.85, .71, 0.0))
# ke_rho=0, ssa=nan, rayleigh=True

print(jax.grad(lambda n: nm.sphere_bulk_optics(
    5.5e-7, 1e-7, 1.8, n, 0.0).ssa)(1.0))
# nan, even though the NETWORK arm is selected at these sizes
```

The final case shows that an invalid discarded Rayleigh arm also contaminates gradients on the network path. The issue is specifically the documented clipping behavior; callers using `clip=False` make a different contract.

Apply clipping before constructing both the sphere Rayleigh indices and the core-shell volume mixture. Test forward equality to explicitly clipped boundary inputs and finite gradients of all three outputs in both branch regimes. Preserve upstream behavior for in-domain inputs and resync the downstream copy.
